src/main/java/org/fundacionjala/coding/rodrigo/movierefactor/Movie.java

```diff
@@ -0,0 +1,67 @@
+package org.fundacionjala.coding.rodrigo.movierefactor;
+
+/**
+ * Abstract class for movie.
+ */
+public abstract class Movie {
+    private String title;
+    private int daysRented;
+    protected double charge;
+
+    /**
+     * Constructor.
+     *
+     * @param title      of the movie.
+     * @param daysRented of the movie.
+     */
+    Movie(final String title, final int daysRented) {
+        this.title = title;
+        this.daysRented = daysRented;
+    }
+
+    /**
+     * Some calc of the charge.
+     */
+    abstract void calculateCharge();
+
+    /**
+     * getter.
+     *
+     * @return title.
+     */
+    public String getTitle() {
+        return title;
+    }
+
+    /**
+     * getter.
+     *
+     * @return MovieType.
+     */
+    public abstract MovieType getMovieType();
```

Comment:
Since we are using abstraction this method is no longer used.
Please remove it!

Reply:
done